llvm-hs/llvm-hs.cabal

```diff
@@ -224,7 +224,7 @@ library
     src/LLVM/Internal/FFI/TypeC.cpp
     src/LLVM/Internal/FFI/ValueC.cpp
 
-  cc-options: -std=c++11 -Wno-stringop-overflow
+  cc-options: -std=c++11 -Wno-stringop-overflow -Wno-init-list-lifetime
```

Comment:
Which parts of the code raise those warnings for you?

Reply:
Unfortunately, these warnings are generated not in our code but in headers included from LLVM. There's more to the story, as usual: for some reason, cabal currently promotes some warnings generated in the compilation of FFI C code to errors, *but then ignores them* and continues the build. So you see a whole bunch of red "error: " messages during the llvm-hs build, and then it completes successfully anyway. If you are using an older version of LLVM to compile the C code, you may not see any warnings.

Since these warnings are generated in LLVM header files, there is not much we can do about them, so I think the simplest thing to do is just silence the warnings so we don't see totally spurious "error" messages from cabal during llvm-hs builds. It may be that the LLVM 12 release branch has fixed these warnings, so we may be able to remove those suppressions on our llvm-12 branch -- I'll check to see if that's the case (#333)